lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -106,159 +103,47 @@ class QcFlagService {
-                attributes: ['timeTrgStart', 'timeTrgEnd'],
-                where: { runNumber },
-                include: [
-                    {
-                        association: 'dataPass',
-                        where: { id: dataPassId },
-                        through: { attributes: [] },
-                        attributes: ['id'],
-                        required: true,
-                    },
-                    {
-                        association: 'detectors',
-                        where: { name: dplDetector.name },
-                        through: { attributes: [] },
-                        attributes: [],
-                        required: true,
-                    },
-                ],
-            });
-            if (!targetRun) {
-                // eslint-disable-next-line max-len
-                throw new BadParameterError(`There is not association between data pass with this id (${dataPassId}), run with this number (${runNumber}) and detector with this name (${dplDetector.name})`);
-            }
-
-            const { fromTime, toTime } = this._prepareQcFlagPeriod({ from, to }, targetRun);
-
-            // Insert
-            const newInstance = await QcFlagRepository.insert({
-                from: fromTime,
-                to: toTime,
-                comment,
-                createdById: user.id,
-                flagTypeId,
-                runNumber,
-                dplDetectorId,
-            });
-
-            const createdFlag = await QcFlagRepository.findOne(this.prepareQueryBuilder().where('id').is(newInstance.id));
-
-            await createdFlag.addDataPasses(targetRun.dataPass);
-
-            return qcFlagAdapter.toEntity(createdFlag);
-        });
-    }
-
-    /**
-     * Create new instance of quality control flags for simulation pass
-     * @param {Partial<QcFlag>} parameters flag instance parameters
-     * @param {object} [relations] QC Flag entity relations
-     * @param {Partial<UserIdentifier>} [relations.user] user identifiers
-     * @param {number} [parameters.flagTypeId] flag type id
-     * @param {number} [parameters.runNumber] associated run's number
-     * @param {number} [parameters.simulationPassId] associated simulationPass' id
-     * @param {number} [parameters.dplDetectorId] associated dplDetector's id
-     * @return {Promise<QcFlag>} promise
-     * @throws {BadParameterError, NotFoundError}
-     */
-    async createForSimulationPass(parameters, relations = {}) {
-        const {
-            from = null,
-            to = null,
-            comment,
-        } = parameters;
-        const {
-            user: { userId, externalUserId } = {},
-            flagTypeId,
-            runNumber,
             simulationPassId,
             dplDetectorId,
         } = relations;
 
+        if (!dataPassId && !simulationPassId) {
+            throw new BadParameterError('Cannot create QC flag without data pass or simulation pass');
+        }
```

Comment:
those parameters should be exclusive (it's not checked later)

Reply:
Indeed, done